### PR TITLE
[GPU] Fix accuracy issue

### PR DESCRIPTION
**Pull request by ahnyoung-paul**
### Details:
 - *[scatter_update] Use input index for input buffer instead of output index*
 - *[concat cpu impl] Sync input layout and mem_ptr when input host tensor creation*
 - *Add unit tests for scatter_update and concat cpu impl*

### Tickets:
 - *100495*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/impls/cpu/concat.cpp [hunk 1]
```diff
@@ -69,17 +69,18 @@ struct concatenation_impl : public typed_primitive_impl<concatenation> {
...
         for (size_t i = 0; i < instance.dependencies().size(); i++) {
             auto& dep = instance.dependencies().at(i);
-            if (dep.first->get_output_layout().count() > 0)
-                input_mem_ptrs.push_back(instance.dep_memory_ptr(i));
+            if (dep.first->get_output_layout().count() > 0) {
+                auto mem_ptr = instance.dep_memory_ptr(i);
+                input_host_tensors.push_back(make_tensor(params->input_layouts[i], mem_ptr->lock(stream, mem_lock_type::read)));
+                // push mem_ptr to input_mem_ptr to unlock after processing
+                input_mem_ptrs.push_back(mem_ptr);
+            }
```
Comment (e-ddykim) [inline src/plugins/intel_gpu/src/graph/impls/cpu/concat.cpp:77]: I don't understand what is changed in this part.

Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/graph/impls/cpu/concat.cpp:77]: When concat has 5 inputs and 3rd input are empty layout, previous code create input_host_tensor with 3rd input layout 4th input memory pointer. I fixed this issue.

Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/impls/cpu/concat.cpp:77]: What I think is that 
when one of the  dep nodes is  empty tensor
then it was not added to input_mem_ptrs. 
e.g.,there are three inputs dep0, dep1, dep2 and dep1 is empty, then input_mem_ptrs has only dep0 and dep2. 
However in the original impl
line 80 is iterating for those two and making host tensor from input_layouts(0) and input_layouts(1) , 
not input_layouts(0) and input_laytous(2)

Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/graph/impls/cpu/concat.cpp:77]: Yes, @yeonbok is right. thanks.

**Review by yeonbok — APPROVED** [review]

File: src/plugins/intel_gpu/src/graph/impls/cpu/concat.cpp (hunk 1 above)
Comment (e-ddykim) [reply src/plugins/intel_gpu/src/graph/impls/cpu/concat.cpp:77]: Oh I see, thank you.

**Review by e-ddykim — APPROVED** [review]

**Pull request merged**